src/averageAges.js

```diff
@@ -55,7 +73,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const womenArr = people.filter(element =>
+    element.sex === 'f' && people.some(child =>
+      child.mother === element.name)
+  );
+
+  const childrenArr = people.filter(element =>
+    onlyWithSon
+      ? element.sex === 'm'
+      && people.some(mother => element.mother === mother.name)
+      : people.some(mother => element.mother === mother.name)
+  );
+
+  const dif = childrenArr.map(element =>
+    element.born - womenArr.find(mother =>
+      element.mother === mother.name).born
+  );
```

Comment:
you can use people and delete get womenArr (-1 cycle). Also, your code is hard to read (

```suggestion
  const dif = childrenArr.map(element => {
    const motherBorn = people.find(
      mother => element.mother === mother.name,
    ).born;
    
    element.born - motherBorn;
  });
```